storm-core/src/jvm/org/apache/storm/task/TopologyContext.java

```diff
@@ -386,4 +388,28 @@ public ReducedMetric registerMetric(String name, IReducer reducer, int timeBucke
     public CombinedMetric registerMetric(String name, ICombiner combiner, int timeBucketSizeInSecs) {
```

Comment:
Can we deprecate the old way of doing these?

Reply:
+1 Let's guide using new metrics if possible.